### DISCUSSION TITLE
incorrect Timer Period on Wemos ESP32S2 Mini & Pico

### DISCUSSION THREAD

**Discussion by fpgirard** [micropython/micropython#10184] [Hardware & Peripherals]
I'm getting an errant period interval on my timers on the Lolin/Wemos ESP32S2 boards (both the Pico & Mini).   It's exactly double the length of time stated in the timer initialization.   The period interval is correct when I flash these boards from the [downloadable micropython site](https://micropython.org/download/LOLIN_S2_PICO/) but when I compile it from scratch, it's 2x the value.  I've git pulled the latest micropython and checked out ESP-IDF 4.4) - `git checkout v4.4` and `git submodule update --init --recursive`.   I've also specified the correct board during the build - `idf.py -D BOARD=LOLIN_S2_PICO build` and erased the board prior to flashing.

Obviously, the easiest thing for me to do is to use the downloadable bin file but I was concerned that I might have discovered something that needs to be fixed for these boards.  On a separate note, Peter Hinch, your asyncio MQTT code is awesome - my modules have been running for weeks on an intermittent Wifi network since I incorporated your MQTT libraries.  Thank you.

Here's the simplest code that reveals the problem:

```
from machine import Timer, enable_irq, disable_irq
import time

publish = False
timer = Timer(0)

def handle_mqtt(timer):
    global publish
    publish = True

timer.init(period=5000, mode=Timer.PERIODIC, callback=handle_mqtt)

def main():
    counter = 0
    global publish
    print("Start timex code.")
    while (True):
        if publish : 
            state = disable_irq()
            publish = False
            enable_irq(state)
            print('Counter is: ', counter)
            counter = 0
        else : 
            time.sleep(1)
            counter += 1
```

**Comment by fivdi** [comment]
Does the MicroPython version you are compiling include this pull request: https://github.com/micropython/micropython/pull/10014?

**Reply by fpgirard** [reply]
It didn't and this was the problem.   Thank you for pointing it out - glad to see that it's been addressed.  I'll pull this request.
